synapse/types/__init__.py

```diff
@@ -77,6 +77,10 @@
 # A JSON-serialisable object.
 JsonSerializable = object
 
+# Collection[str] that does not include str itself; str being a Sequence[str]
+# is very misleading and results in bugs.
+StrCollection = Union[Tuple[str, ...], List[str], Set[str]]
```

Comment:
❤️

Reply:
it was about time someone did it :D

Reply:
Filed #14809 about trying to use this in more places.